Source/ARTRestChannel.m

```diff
@@ -214,6 +218,57 @@ - (BOOL)history:(ARTDataQuery *)query callback:(ARTPaginatedMessagesCallback)cal
     return ret;
 }
 
+- (void)status:(ARTChannelDetailsCallback)callback {
+    if (callback) {
+        ARTChannelDetailsCallback userCallback = callback;
+        callback = ^(ARTChannelDetails *details, ARTErrorInfo *_Nullable error) {
+            dispatch_async(self->_userQueue, ^{
+                userCallback(details, error);
+            });
+        };
+    }
+    dispatch_async(_queue, ^{
+        NSURL *url = [NSURL URLWithString:self->_basePath];
+        NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:url];
+        
+        [self.logger debug:__FILE__ line:__LINE__ message:@"RS:%p C:%p (%@) channel details request %@", self->_rest, self, self.name, request];
+        
+        [self->_rest executeRequest:request withAuthOption:ARTAuthenticationOn completion:^(NSHTTPURLResponse * _Nullable response, NSData * _Nullable data, NSError * _Nullable error) {
+            
+            if (response.statusCode == 200 /*OK*/) {
+                NSError *decodeError = nil;
+                ARTChannelDetails *channelDetails = [[self->_rest defaultEncoder] decodeChannelDetails:data error:&decodeError];
```

Comment:
Instead of using `defaultEncoder`, should we use the MIME type of the response to make sure we use the correct encoder depending on whether we receive JSON or MessagePack in the response? See [this example of `-[ARTRest time:]`](https://github.com/ably/ably-cocoa/blob/0fe3c59658ce7d0b7181571cdad1f068d4085088/Source/ARTRest.m#L522).

Reply:
https://github.com/ably/ably-cocoa/pull/1420/commits/f83d845dd5f51e680acec4a04a323eabbeb356a0

Reply:
Cool, I've just noticed we should probably also set the `Accept` header on the request, [again similarly to `-[ARTRest time:]`](https://github.com/ably/ably-cocoa/blob/0fe3c59658ce7d0b7181571cdad1f068d4085088/Source/ARTRest.m#L509-L510).

Also, I just noticed another issue with `-[ARTRest time:]` which I think applies here, too: https://github.com/ably/ably-cocoa/issues/1442. If this is the case, please could you make an issue (or update the existing one) so that we address it here too?